examples/minimal/packages/contracts/foundry.toml

```diff
@@ -2,7 +2,7 @@
 solc_version = "0.8.13"
 ffi = false
 fuzz_runs = 256
-optimizer = true
+optimizer = false
```

Comment:
Set this to `true` and `SyncSystem` fails to compile. Something weird about 2D arrays: https://github.com/ethereum/solidity/issues/11638

Reply:
Yes, definitely - we should NOT ship this. I need to look into it but wanted to get the code working

it might actually be fine now the function signature has changed